# HG changeset patch
# User Sergei Trofimovich <slyfox@gentoo.org>
# Date 1304189648 -10800
# Node ID 7d6d8bcbaa9b350f36556d985cd291a58aeec4d7
# Parent  63c47842eb39e7ed587d0906880f61651e8533df
fixed build breakage on not-so-old ghc-6.12.3 (Control.Monad.void)

[ 7 of 11] Compiling Criterion.Plot   ( Criterion/Plot.hs, dist/build/Criterion/Plot.o )

Criterion/Plot.hs:21:22:
    Module `Control.Monad' does not export `void'

diff -r 63c47842eb39 -r 7d6d8bcbaa9b Criterion/Plot.hs
--- a/Criterion/Plot.hs	Mon Apr 18 11:44:09 2011 -0700
+++ b/Criterion/Plot.hs	Sat Apr 30 21:54:08 2011 +0300
@@ -18,7 +18,6 @@
     , plotWith
     ) where
 
-import Control.Monad (void)
 import Control.Monad.Trans (liftIO)
 import Criterion.Config
 import Criterion.Monad (Criterion, getConfigItem)
@@ -43,6 +42,13 @@
 #endif
 #endif
 
+#ifdef HAVE_CHART
+-- | introduced in ghc-7. keep a copy here to build against ghc-6.12
+-- | @'void' value@ discards or ignores the result of evaluation, such as the return value of an 'IO' action.
+void :: Functor f => f a -> f ()
+void = fmap (const ())
+#endif
+
 plotWith :: Plot -> (PlotOutput -> IO ()) -> Criterion ()
 plotWith p plot = getConfigItem (M.lookup p . cfgPlot)
                     >>= maybe (return ()) (liftIO . flip forM_ plot)
